src/Cohesity.Powershell/Scripts/Utility/Generate-CohesityPSCmdletTemplate.ps1

```diff
@@ -0,0 +1,167 @@
+            # Creating Process block for template file with body payload part for Post, Put, Register operation
+            $processBlock = @"
+
+        #       `$payload = @{}
+        #       `$payloadJson = $payload | ConvertTo-Json -Depth 100
+        #       `$resp = Invoke-RestApi -Method $httpMethod -Uri `$cohesityUrl -Headers `$cohesityHeaders -Body `$payloadJson
+"@
+        }
+
+        if($MethodType -eq "get" -OR $MethodType -eq "delete"){
+            # Creating Process block for template file without body payload part for Get and Delete operation
+            $processBlock = @"
+
+        #       `$resp = "Invoke-RestApi -Method $httpMethod -Uri `$cohesityUrl -Headers `$cohesityHeaders"
+"@
+        }
+        
+        # Creating the template body here.
+        $FileContent = @"
+function $filePrefix-Cohesity$Feature {
+    <#
+        .SYNOPSIS
+        <string>.
+        .DESCRIPTION
+        <string>.
+        .NOTES
+        <string>
+        <Add any specific note for the cmdlet or any prerequisites>
+        .LINK
+        https://cohesity.github.io/cohesity-powershell-module/#/README
+        .EXAMPLE
+        $filePrefix-Cohesity$Feature -Param1 <string>
+    #>
+    [CmdletBinding($(if($MethodType -eq "delete"){'SupportsShouldProcess = $True, ConfirmImpact = "High"'}))]
+    Param(
+        [Parameter(Mandatory = `$false)]
+        `$Param1
+    )
+
+    Begin {
+        if (-not (Test-Path -Path "$HOME/.cohesity")) {
+            throw "Failed to authenticate. Please connect to the Cohesity Cluster using 'Connect-CohesityCluster'"
+        }
+        `$cohesitySession = Get-Content -Path $HOME/.cohesity | ConvertFrom-Json
+        `$cohesityServer = `$cohesitySession.ClusterUri
+        `$cohesityToken = `$cohesitySession.Accesstoken.Accesstoken
+    }
+
+    Process {
+        #       Please uncomment the code to use it
+        
+        #       Append the url with your MethodType accordingly.
+        #       `$cohesityUrl = `$cohesityServer + '/irisservices/api/v1/public/'
+        #       `$cohesityHeaders = @{'Authorization' = 'Bearer ' + `$cohesityToken }
+                $(if($MethodType -eq "delete") {             
+        '#      if($PSCmdlet.ShouldProcess($Param1)){'
+                })
+                $processBlock
+
+        #       if (`$resp) {
+        #           `$resp
+        #       }
+        #       else {
+        #           `$errorMsg = $errorMessage
+        #           Write-Host `$errorMsg
+        #           CSLog -Message `$errorMsg
+        #       }
+
+                $(if($MethodType -eq "delete") {                
+        
+        '#      }
+        #       else {
+        #              return
+        #       }'
+                })      
+    }
+
+    End { 
+    }
+}
+"@ | out-file $fileLoc # Writing the content of the template into the new created file.   
+    }
+
+    End { 
+    } 
+} 
```

Comment:
As discussed, lets put this in tools folder instead of making it a commandlet

Reply:
Will push the script in tool folder outside the "Script" folder